### PR TITLE
docs: Add labels and annotations to created namespaces

### PR DESCRIPTION
**Pull request by philnichol**
Thanks in advance for your review.

This adds the expected labels and annotations to the namespaces that automatically come with a helm installation.

Closes https://github.com/linkerd/linkerd2/issues/7298

I agree to the DCO for all the commits in this PR.

### CODE REVIEW

File: linkerd.io/content/2.10/tasks/automatically-rotating-webhook-tls-credentials.md [hunk 1]
```diff
@@ -34,12 +34,22 @@ namespace linkerd uses:
...
+  config.linkerd.io/admission-webhooks=disabled \
+  linkerd.io/control-plane-ns=linkerd
+kubectl annotate namespace linkerd linkerd.io/inject=disabled
 
 # viz (ignore if not using the viz extension)
 kubectl create namespace linkerd-viz
+kubectl label namespace linkerd-viz linkerd.io/extension=viz
+kubectl annotate namespace linkerd-viz \
+  config.linkerd.io/proxy-await=enabled \
+  linkerd.io/inject=enabled
```
Comment (alpeb) [inline linkerd.io/content/2.10/tasks/automatically-rotating-webhook-tls-credentials.md:48]: These annotations are not required

**Review by alpeb — COMMENTED** [review]
Thanks for the PR @philnichol , and sorry for the late review!

I think we shouldn't push these changes to the 2.9 and 2.10 docs given they might have had different requirements. OTOH would you mind applying them as well to the 2.12 docs that have been merged into this branch?

**Comment by philnichol** [comment]
Hey @alpeb no problem at all and thanks for your review.
I've removed the unneeded annotations, reverted the 2.9 and 2.10 docs, and added this to 2.12.
Any issues please let me know.

**Review by alpeb — APPROVED** [review]
Looks good, thanks @philnichol !

**Pull request merged**